js/index.js

```diff
@@ -103,19 +103,37 @@ function outputMessage(accepted) {
 
 const form = document.querySelector('.container-form');
 
+const email = document.getElementById('email');
+const name = document.getElementById('name');
+const message = document.getElementById('message');
+
 form.addEventListener('submit', (element) => {
-  const email = document.getElementById('email').value;
   const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
 
-  if (email !== email.toLowerCase()) {
+  if (email.value !== email.value.toLowerCase()) {
     element.preventDefault();
     outputMessage(false);
     return;
   }
-  if (!emailRegex.test(email)) {
+  if (!emailRegex.test(email.value)) {
     element.preventDefault();
     outputMessage(false);
     return;
   }
   outputMessage(true);
 });
+
+window.onload = () => {
+  const showData = JSON.parse(localStorage.getItem('dataform'));
+
+  if (showData) {
+    email.value = showData.email;
+    name.value = showData.name;
+    message.value = showData.message;
+  }
+};
+
+form.addEventListener('input', () => {
+  const data = { email: email.value, name: name.value, message: message.value };
+  localStorage.setItem('dataform', JSON.stringify(data));
```

Comment:
- [ ] [OPTIONAL] Good job implementing the suggestion of the previous reviewer, but I strongly suggest using the `input` eventListener for an `input`, `textarea` or `select` tag. You can read more about it [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event)